example/convolution2D/src/convolution2D.cpp

```diff
@@ -0,0 +1,381 @@
+                                  * sharedN[exactRowBlock * blockThreadExtentX + exactColBlock];
+                    }
+                    else
+                    { // The element is not in the tile(block)
+                        // Position of input matrix element to be multiplied with the corresponding element at the
+                        // filter. The position is with respect to the input matrix
+                        auto const exactRow = static_cast<int32_t>(row) - filterWidth / 2 + fRow;
+                        auto const exactCol = static_cast<int32_t>(col) - filterWidth / 2 + fCol;
+                        if(exactRow >= 0 && exactRow < matrixHeight && exactCol >= 0 && exactCol < matrixWidth)
+                        {
+                            // get the item from the global memory, use padded width!
+                            pValue += filter[fRow * filterWidthAllocated + fCol]
+                                      * input[exactRow * intputWidthAllocated + exactCol];
+                        }
+                    }
+                }
+                output[row * matrixWidth + col] = pValue;
+            }
+        } // if
+    }
+};
+
+// The specialisation used for calculation of dynamic shared memory size
+namespace alpaka::trait
+{
+    //! The trait for getting the size of the block shared dynamic memory for a kernel.
+    template<typename TAcc>
+    struct BlockSharedMemDynSizeBytes<ConvolutionKernel2DSharedMemory, TAcc>
+    {
+        //! \return The size of the shared memory allocated for a block.
+        template<typename TVec, typename TElem>
+        ALPAKA_FN_HOST_ACC static auto getBlockSharedMemDynSizeBytes(
+            ConvolutionKernel2DSharedMemory const& /* matMulKernel */,
+            TVec const& blockThreadExtent, // dimensions of thread per block
+            TVec const& threadElemExtent, // dimensions of elements per thread
+            TElem const* const, // input Matrix
+            TElem*, // output array
+            int32_t const, // matrixWidth
+            int32_t const, // matrixHeight
+            TElem const* const, // filter
+            int32_t const, // filter width
+            int32_t const, // allocated input width
+            int32_t const) // allocated filter width
+        {
+            // Reserve the buffer, buffers size is the number of elements in a block (tile)
+            return static_cast<std::size_t>(1u * blockThreadExtent.prod() * threadElemExtent.prod()) * sizeof(TElem);
+        }
+    };
+} // namespace alpaka::trait
+
+auto FuzzyEqual(float a, float b) -> bool
+{
+    return std::fabs(a - b) < std::numeric_limits<float>::epsilon() * 1000.0f;
+}
+
+auto main() -> int
+{
+    // Define the index domain
+    using Dim = alpaka::DimInt<2>;
+    // Index type
+    using Idx = std::uint32_t;
+    using Vec = alpaka::Vec<Dim, Idx>;
+    // Define the accelerator
+    using DevAcc = alpaka::AccCpuSerial<Dim, Idx>;
```

Comment:
Let us have an offline chat, I am wondering why you use the serial accelerator here. We should not write examples only for CPU or only for GPU. 
An example should run on all platforms.

Reply:
This was for a test run. I changed to _ExampleDefaultAcc_. Thanks.